crates/evm/evm/src/executors/fuzz/mod.rs

```diff
@@ -98,7 +102,17 @@ impl FuzzedExecutor {
         let max_traces_to_collect = std::cmp::max(1, self.config.gas_report_samples) as usize;
         let show_logs = self.config.show_logs;
 
+        // Start a timer if timeout is set.
+        let start_time = start_timer(self.config.timeout);
+
         let run_result = self.runner.clone().run(&strategy, |calldata| {
+            // Check if the timeout has been reached.
+            if let Some((start_time, timeout)) = start_time {
+                if start_time.elapsed() > timeout {
```

Comment:
I guess since this is opt-in, checking via elapsed for now is fine, and we can track improvements separately, but I'd suggest to wrap this if let Some into a helper type like `Timeout(Option<Instant>)` and an `is_timed_out` then this becomes easier to change

Reply:
@mattsse I added `FuzzTestTimer` with https://github.com/foundry-rs/foundry/pull/9394/commits/da87095e1fe7f7cb7de4d8099c2307ccd408982b please check